zookeeper-docs/src/main/resources/markdown/zookeeperUseCases.md

```diff
@@ -0,0 +1,319 @@
+    the library itself is designed to detect and handle failures at the application layer, so delivering a highly-available service on top of a cluster of computers, each of which may be prone to failures.
+  - The implementation of automatic HDFS failover relies on ZooKeeper for the following things:
+    - **Failure detection** - each of the NameNode machines in the cluster maintains a persistent session in ZooKeeper.
+      If the machine crashes, the ZooKeeper session will expire, notifying the other NameNode that a failover should be triggered.
+    - **Active NameNode election** - ZooKeeper provides a simple mechanism to exclusively elect a node as active. If the current active NameNode crashes,
+      another node may take a special exclusive lock in ZooKeeper indicating that it should become the next active.
+  - The ZKFailoverController (ZKFC) is a new component which is a ZooKeeper client which also monitors and manages the state of the NameNode.
+    Each of the machines which runs a NameNode also runs a ZKFC, and that ZKFC is responsible for:
+    - **Health monitoring** - the ZKFC pings its local NameNode on a periodic basis with a health-check command.
+      So long as the NameNode responds in a timely fashion with a healthy status, the ZKFC considers the node healthy.
+      If the node has crashed, frozen, or otherwise entered an unhealthy state, the health monitor will mark it as unhealthy.
+    - **ZooKeeper session management** - when the local NameNode is healthy, the ZKFC holds a session open in ZooKeeper.
+      If the local NameNode is active, it also holds a special “lock” znode. This lock uses ZooKeeper’s support for “ephemeral” nodes;
+      if the session expires, the lock node will be automatically deleted.
+    - **ZooKeeper-based election** - if the local NameNode is healthy, and the ZKFC sees that no other node currently holds the lock znode,
+      it will itself try to acquire the lock. If it succeeds, then it has “won the election”, and is responsible for running a failover to make its local NameNode active.
+      The failover process is similar to the manual failover described above: first, the previous active is fenced if necessary,
+      and then the local NameNode transitions to active state.[7]
+
+### [Apache HBase](https://hbase.apache.org/)
+  - HBase is the Hadoop database. It's an open-source, distributed, column-oriented store model.
+  - HBase uses ZooKeeper for master election, server lease management, bootstrapping, and coordination between servers.
+    A distributed Apache HBase installation depends on a running ZooKeeper cluster. All participating nodes and clients
+    need to be able to access the running ZooKeeper ensemble.[8]
+  - As you can see, ZooKeeper is a fundamental part of HBase. All operations that require coordination, such as Regions
+    assignment, Master-Failover, replication, and snapshots, are built on ZooKeeper[20].
+
+### [Apache Helix](http://helix.apache.org/)
+
```

Comment:
the abstract of this project is missing here.

Reply:
- Now this PR re-enters a state of reviewing.
- I'am always be here to waiting whether someone wants to add other additions.